src/odemis/gui/comp/overlay/world.py

```diff
@@ -832,6 +833,566 @@ def deactivate(self):
+            pos = Vec(l_pos[0], l_pos[1])
+            self.ruler_label.pos = pos
+            self.ruler_label.weight = cairo.FONT_WEIGHT_NORMAL
+
+            # If the ruler is smaller than 1 pixel, make it seem as 1 point (1 pixel) and decrease the font size to 5pt.
+            # Only the move area of the ruler is available, without the option of editing the start, end positions.
+            if pixel_length <= 1:
+                ctx.move_to(*b_start)
+                ctx.line_to(b_start[0] + 1, b_start[1] + 1)
+                ctx.stroke()
+                self.ruler_label.font_size = 5
+                self.pixel_ruler = True
+            else:
+                if pixel_length < 40: # about the length of the label
+                    self.ruler_label.font_size = 9
+                else:
+                    self.ruler_label.font_size = 13
+                    self.ruler_label.weight = cairo.FONT_WEIGHT_BOLD
+                ctx.move_to(*b_start)
+                ctx.line_to(*b_end)
+                ctx.stroke()
+                self.pixel_ruler = False
+
+            # Label and ruler have the same colour, highlighted or not
+            if selected:
+                self.ruler_label.colour = self.highlight
+            else:
+                self.ruler_label.colour = self.colour
+
+            Label.draw(self.ruler_label, ctx)
+            # self._debug_ruler_edges(ctx, True)
+
+
+MODE_CREATE_RULER = 1
+MODE_SHOW_RULER = 2
+
+class RulerSelectOverlay(WorldOverlay, DragMixin):
+    """
+       Selection overlay that allows for the selection of a ruler in physical coordinates
+       and displays the size of the ruler in meters. It can handle multiple rulers.
+    """
+
+    def __init__(self, cnvs, tool_va=None):
+        WorldOverlay.__init__(self, cnvs)
+        DragMixin.__init__(self)
+
+        self.ruler_mode = None
+        self.mode = None
+
+        self._selected_tool_va = tool_va
+        if self._selected_tool_va:
+            self._selected_tool_va.subscribe(self._on_tool, init=True)
+
+        self._selected_ruler = None
+        self._rulers = []  # Rulers and all
+
+    def clear(self):
+        """Remove rulers and update canvas"""
+        self._rulers = []
+        self.cnvs.update_drawing()
+
+
+    def _on_tool(self, selected_tool):
+        """ Update the overlay when it's active and ruler changes"""
+
+        if selected_tool == TOOL_RULER:
+            self.ruler_mode = MODE_CREATE_RULER
+        else:
+            self.ruler_mode = MODE_SHOW_RULER
+
+    def on_left_down(self, evt):
+        """ Start drawing a ruler if the create mode is active, otherwise start editing/moving a selected ruler"""
+
+        if not self.active:
+            return super(RulerSelectOverlay, self).on_left_down(evt)
+
+        if self.right_dragging:
+            evt.Skip()
+            return
+
+        vpos = evt.Position
+        self.drag_v_start_pos = self.drag_v_end_pos = Vec(vpos)
```

Comment:
`self.drag_v_start_pos` and `self.drag_v_end_pos` don't seem used outside of this method. With DragMixin, it's a little more confusing, but I have a feeling that it's still not used. Actually, I have a feeling this class doesn't really need to inherit from DragMixin, as it just needs a couple of similar code.